### ISSUE TITLE
Generate conda packages CI failure Begin June 2022

### ISSUE THREAD

**Issue by traversaro** [robotology/robotology-superbuild#1140]
The generation (https://github.com/robotology/robotology-superbuild/runs/6662595195) failed with :
~~~
2022-05-31T01:23:41.0442900Z Glob /home/runner/work/robotology-superbuild/robotology-superbuild/build/conda/generated_recipes/yarp-telemetry did not match in root_dir /home/runner/work/robotology-superbuild/robotology-superbuild/build/conda/generated_recipes
2022-05-31T01:23:41.0444060Z Updating build index: /usr/share/miniconda/envs/test/conda-bld
2022-05-31T01:23:41.0444590Z 
2022-05-31T01:23:41.0446178Z ERROR:conda_build.utils:Glob /home/runner/work/robotology-superbuild/robotology-superbuild/build/conda/generated_recipes/yarp-telemetry did not match in root_dir /home/runner/work/robotology-superbuild/robotology-superbuild/build/conda/generated_recipes
2022-05-31T01:23:41.0457032Z Traceback (most recent call last):
2022-05-31T01:23:41.0457676Z   File "/usr/share/miniconda/envs/test/bin/conda-mambabuild", line 10, in <module>
2022-05-31T01:23:41.0458289Z     sys.exit(main())
2022-05-31T01:23:41.0458947Z   File "/usr/share/miniconda/envs/test/lib/python3.8/site-packages/boa/cli/mambabuild.py", line 215, in main
2022-05-31T01:23:41.0459455Z     call_conda_build(action, config)
2022-05-31T01:23:41.0460095Z   File "/usr/share/miniconda/envs/test/lib/python3.8/site-packages/boa/cli/mambabuild.py", line 187, in call_conda_build
2022-05-31T01:23:41.0460800Z     result = api.build(
2022-05-31T01:23:41.0461443Z   File "/usr/share/miniconda/envs/test/lib/python3.8/site-packages/conda_build/api.py", line 184, in build
2022-05-31T01:23:41.0461979Z     raise ValueError(f'No valid recipes found for input: {recipe_paths_or_metadata}')
2022-05-31T01:23:41.0462424Z ValueError: No valid recipes found for input: yarp-telemetry
~~~

This is fixed by https://github.com/traversaro/robotology-superbuild/commit/ab2126f06e10ce83b59c05e4087735201ebf4e22 . However, even with that fix the generation is failing (https://github.com/traversaro/robotology-superbuild/actions/runs/2412094914). On Linux/macOS, the generation is failing with: 
~~~
2022-05-31T03:18:04.7191727Z Adding in variants from internal_defaults
2022-05-31T03:18:04.7195274Z INFO:conda_build.variants:Adding in variants from /usr/share/miniconda/envs/test/conda_build_config.yaml
2022-05-31T03:18:04.7195792Z Adding in variants from /usr/share/miniconda/envs/test/conda_build_config.yaml
2022-05-31T03:18:04.7198838Z INFO:conda_build.variants:Adding in variants from /home/runner/work/robotology-superbuild/robotology-superbuild/conda/conda_build_config.yml
2022-05-31T03:18:04.7200039Z Adding in variants from /home/runner/work/robotology-superbuild/robotology-superbuild/conda/conda_build_config.yml
2022-05-31T03:18:06.4724767Z INFO:conda_build.metadata:Attempting to finalize metadata for bipedal-locomotion-framework
2022-05-31T03:18:06.4725476Z Attempting to finalize metadata for bipedal-locomotion-framework
2022-05-31T03:18:06.6251897Z conda-forge/linux-64     Using cache
2022-05-31T03:18:06.6273260Z conda-forge/noarch       Using cache
2022-05-31T03:18:06.9316038Z Reloading output folder: /usr/share/miniconda/envs/test/conda-bld
2022-05-31T03:18:07.8299026Z Mamba failed to solve:
2022-05-31T03:18:07.8300025Z  - lie-group-controllers
2022-05-31T03:18:07.8300878Z  - manif
2022-05-31T03:18:07.8301167Z  - pybind11
2022-05-31T03:18:07.8301405Z  - idyntree
2022-05-31T03:18:07.8301640Z  - casadi
2022-05-31T03:18:07.8301842Z  - eigen
2022-05-31T03:18:07.8302068Z  - matio-cpp
2022-05-31T03:18:07.8302307Z  - numpy 1.19.*
2022-05-31T03:18:07.8302709Z  - xorg-libxfixes
2022-05-31T03:18:07.8302984Z  - pybind11-abi
2022-05-31T03:18:07.8303215Z  - python
2022-05-31T03:18:07.8303433Z  - manifpy
2022-05-31T03:18:07.8303661Z  - pybind11-abi 4.*
2022-05-31T03:18:07.8303963Z  - unicycle-footstep-planner
2022-05-31T03:18:07.8304240Z  - osqp-eigen
2022-05-31T03:18:07.8304460Z  - yarp
2022-05-31T03:18:07.8304680Z  - yarp-telemetry
2022-05-31T03:18:07.8304908Z  - qhull
2022-05-31T03:18:07.8305137Z  - nlohmann_json
2022-05-31T03:18:07.8305349Z  - spdlog
2022-05-31T03:18:07.8305593Z  - python 3.7.* *_cpython
2022-05-31T03:18:07.8305843Z  - spdlog 1.10.*
2022-05-31T03:18:07.8306069Z  - numpy
2022-05-31T03:18:07.8306271Z  - cppad
2022-05-31T03:18:07.8306386Z 
2022-05-31T03:18:07.8306471Z with channels:
2022-05-31T03:18:07.8306599Z 
2022-05-31T03:18:07.8306697Z The reported errors are:
2022-05-31T03:18:07.8307018Z - Encountered problems while solving:
2022-05-31T03:18:07.8307596Z -   - package unicycle-footstep-planner-0.4.1-h27087fc_54 requires idyntree >=5.2.1,<6.0a0, but none of the providers can be installed
2022-05-31T03:18:07.8307992Z - 
2022-05-31T03:18:07.8308202Z usr/share/miniconda/envs  
2022-05-31T03:18:07.8308437Z usr/share/miniconda/envs  
2022-05-31T03:18:07.8389554Z Traceback (most recent call last):
2022-05-31T03:18:07.8391396Z   File "/usr/share/miniconda/envs/test/lib/python3.8/site-packages/boa/cli/mambabuild.py", line 123, in mamba_get_install_actions
2022-05-31T03:18:07.8392626Z     solution = solver.solve_for_action(_specs, prefix)
2022-05-31T03:18:07.8393248Z   File "/usr/share/miniconda/envs/test/lib/python3.8/site-packages/boa/core/solver.py", line 234, in solve_for_action
2022-05-31T03:18:07.8393575Z     t = self.solve(specs)
2022-05-31T03:18:07.8394041Z   File "/usr/share/miniconda/envs/test/lib/python3.8/site-packages/boa/core/solver.py", line 216, in solve
2022-05-31T03:18:07.8394697Z     raise RuntimeError("Solver could not find solution." + error_string)
2022-05-31T03:18:07.8395062Z RuntimeError: Solver could not find solution.Mamba failed to solve:
2022-05-31T03:18:07.8395414Z  - lie-group-controllers
2022-05-31T03:18:07.8395670Z  - manif
2022-05-31T03:18:07.8395898Z  - pybind11
2022-05-31T03:18:07.8396126Z  - idyntree
2022-05-31T03:18:07.8396329Z  - casadi
2022-05-31T03:18:07.8396540Z  - eigen
2022-05-31T03:18:07.8396760Z  - matio-cpp
2022-05-31T03:18:07.8396983Z  - numpy 1.19.*
2022-05-31T03:18:07.8397226Z  - xorg-libxfixes
2022-05-31T03:18:07.8397467Z  - pybind11-abi
2022-05-31T03:18:07.8397692Z  - python
2022-05-31T03:18:07.8397896Z  - manifpy
2022-05-31T03:18:07.8398133Z  - pybind11-abi 4.*
2022-05-31T03:18:07.8398433Z  - unicycle-footstep-planner
2022-05-31T03:18:07.8398701Z  - osqp-eigen
2022-05-31T03:18:07.8398911Z  - yarp
2022-05-31T03:18:07.8399145Z  - yarp-telemetry
2022-05-31T03:18:07.8399370Z  - qhull
2022-05-31T03:18:07.8399582Z  - nlohmann_json
2022-05-31T03:18:07.8399810Z  - spdlog
2022-05-31T03:18:07.8400051Z  - python 3.7.* *_cpython
2022-05-31T03:18:07.8400339Z  - spdlog 1.10.*
2022-05-31T03:18:07.8400551Z  - numpy
2022-05-31T03:18:07.8400892Z  - cppad
2022-05-31T03:18:07.8401010Z 
2022-05-31T03:18:07.8401093Z with channels:
2022-05-31T03:18:07.8401225Z 
2022-05-31T03:18:07.8401322Z The reported errors are:
2022-05-31T03:18:07.8401619Z - Encountered problems while solving:
2022-05-31T03:18:07.8402216Z -   - package unicycle-footstep-planner-0.4.1-h27087fc_54 requires idyntree >=5.2.1,<6.0a0, but none of the providers can be installed
~~~

The strange thing is it that it does not list anything in "channels", even because `unicycle-footstep-planner` with idyntree 5.2.1 was just compiled earlier.

On Windows, the compilation of `yarp-device-openxrheadset` is failing with:
~~~
2022-05-31T06:25:18.6048003Z CMake Error at C:/Miniconda/envs/test/conda-bld/yarp-device-openxrheadset_1653978113880/_build_env/Library/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
2022-05-31T06:25:18.6049032Z   Could NOT find GLFW3 (missing: GLFW3_LIBRARIES)
2022-05-31T06:25:18.6049542Z Call Stack (most recent call first):
2022-05-31T06:25:18.6050316Z   C:/Miniconda/envs/test/conda-bld/yarp-device-openxrheadset_1653978113880/_build_env/Library/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
2022-05-31T06:25:18.6052774Z   C:/Miniconda/envs/test/conda-bld/yarp-device-openxrheadset_1653978113880/_h_env/Library/share/YCM/modules/StandardFindModule.cmake:224 (find_package_handle_standard_args)
2022-05-31T06:25:18.6053561Z   C:/Miniconda/envs/test/conda-bld/yarp-device-openxrheadset_1653978113880/_h_env/Library/share/YCM/find-modules/FindGLFW3.cmake:33 (standard_find_module)
2022-05-31T06:25:18.6054163Z   CMakeLists.txt:90 (find_package)
~~~

**Comment by traversaro** [comment]
> The strange thing is it that it does not list anything in "channels", even because `unicycle-footstep-planner` with idyntree 5.2.1 was just compiled earlier.

I tried to switch to `conda build` or explicitly list the channels to use, but in both cases the problem is not fixed. At this point, we will release 2022.05.0 without conda packages, and work on this for then release 2022.05.1 .

**Comment by traversaro** [comment]
On the Windows side, this is related to https://github.com/robotology/robotology-superbuild/issues/1094 .

**Comment by traversaro** [comment]
We were quite lucky, the conda packages in https://github.com/robotology/robotology-superbuild/actions/runs/2469987668 were generated correctly even in Windows, and the metapackages were the only one not generated, but we can generated those manually.

**Comment by traversaro** [comment]
> We were quite lucky, the conda packages in https://github.com/robotology/robotology-superbuild/actions/runs/2469987668 were generated correctly even in Windows, and the metapackages were the only one not generated, but we can generated those manually.

Unfortunatly the packages were uploaded only after metapackages generation, so no Windows packages were uploaded. Better to move YARP packages to conda-forge to definitely solve https://github.com/robotology/robotology-superbuild/issues/1094 .

**Comment by traversaro** [comment]
The generation is finally working fine: https://github.com/robotology/robotology-superbuild/runs/6871865513 . However, for some reason now the generation **without YARP** takes more time then **with YARP**. However, we can understand this in a new issue.
